### ISSUE TITLE
Add the Refit Exception Content to Sentry

### ISSUE THREAD

**Issue by TimVerheul** [getsentry/sentry-dotnet#3314]
### Problem Statement

Refit is a populair library for making requests. If an exception occurs, only the Exception Message will be sent to Sentry. The Message is a very broad description (a 400 was received) without containing valuable information. 
This information is stored in the Content part of the Exception. Unfortunately, Sentry does not send this.
So when a Refit Exception occurs, you won't know what exactly is causing this.

![image](https://github.com/getsentry/sentry-dotnet/assets/29889196/afa9fd5a-9017-46d9-a836-ab58342f5c42)

### Solution Brainstorm

As a temporary solution I add my own EventExceptionProcessor but it would be nice to have something like .CatchRefit() to handle this.

**Comment by jamescrosswell** [comment]
Hey @TimVerheul , thanks for reaching out. 

There are so many libraries out there and only so many hours in a day, so I'm not sure if can add something specifically for Refit

However, I'm wondering if we could potentially create something a bit more generic. Maybe some way to configure the `SentryHttpFailedRequestHandler` to check for (and extract) JSON content from response bodies for certain request targets and/or HttpStatus codes. 

@bitsandfoxes thoughts?

**Comment by bitsandfoxes** [comment]
This originates from a [discussion on Discord](https://discord.com/channels/621778831602221064/1230283694351057057).

We have a way that enables users to tailor the exception handling to their need if we lack some sort of integration with any specific framework. The way @TimVerheul managed to unblock himself looks like this

```
public class RefitExceptionProcessor : ISentryEventExceptionProcessor
{
    public void Process(Exception exception, SentryEvent sentryEvent)
    {
        if (exception is Refit.ApiException apiException)
        {
            sentryEvent.Message = apiException.Content;
            sentryEvent.Contexts["Refit"] = new
            {
                RefitMessage = apiException.Message,
                RefitContent = apiException.Content,
            }
        }
    }
}
```

I'm not sure about the benefit of crawling through whatever a framework/library has done to their own exception types and "guess" what bits might be relevant.

**Comment by bitsandfoxes** [comment]
I see I misread your comment. You wanted to check the content on the response bodies. Won't we run into all kinds of PII and therefor opt-in/out related issues when trying to parse and capture the bodies?

**Comment by TimVerheul** [comment]
I think having the code written here will help others who might got the same issue as I had. I agree, it's not possible to handle every custom exception for every library.

**Comment by jamescrosswell** [comment]
I think there's a big enough community for Refit that we could look at putting a Sample in the repo as a first step.

<img width="886" height="609" alt="Image" src="https://github.com/user-attachments/assets/14808e88-a180-42c5-b9b3-10815682c7ff" />

> [!TIP]
> Possibly relates to:
> - https://github.com/getsentry/sentry-dotnet/issues/2140
